File: tier4_expert/greedy/problems/p502_ipo.py
"""
P502: IPO [PREMIUM] (Hard)
https://leetcode.com/problems/ipo/
Topics: Array, Greedy, Sorting, Heap (Priority Queue)

Given k, w, profits[], capital[], return maximized capital after at most k projects.

Constraints:
    - 1 <= k <= 10^5
    - 0 <= w <= 10^9
    - 1 <= n <= 10^5
"""

from src.wasm_libs.py.io import *


def solve(k: int, w: int, profits: list[int], capital: list[int]) -> int:
    raise NotImplementedError


if __name__ == "__main__":
    k = read_int()
    w = read_int()
    profits = read_ints()
    capital = read_ints()
    result = solve(k, w, profits, capital)
    write_int(result)

File: tier4_expert/greedy/problems/p502_ipo.rs
/*
 * P502: IPO [PREMIUM] (Hard)
 * https://leetcode.com/problems/ipo/
 * Topics: Array, Greedy, Sorting, Heap (Priority Queue)
 *
 * Given k, w, profits[], capital[], return maximized capital after at most k projects.
 */

use wasm_libs::*;
use std::collections::BinaryHeap;

fn find_maximized_capital(k: i32, mut w: i32, profits: &[i32], capital: &[i32]) -> i32 {
    todo!();
}

fn main() {
    let k = read_int();
    let w = read_int();
    let profits = read_ints();
    let capital = read_ints();
    let result = find_maximized_capital(k, w, &profits, &capital);
    write_int(result);
}
